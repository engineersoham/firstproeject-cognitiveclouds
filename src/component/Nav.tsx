import { AppBar, Button, TextField, Toolbar, Typography } from '@mui/material';
import React, { useState } from 'react'

type Props = {
    navigate:any
}
const Nav:React.FC<Props> = ({...props}) => {

    const [input, setInput ]:any = useState('');

    const handelChange = (e:any)=>{
        setInput(e.target.value)
    }

    const handelSubmit = (e:any)=>{
        e.preventDefault();
        setInput('');
        props.navigate(`/country/${input}`)
    }

  return (
    <div>
        <AppBar position='static'>
            <Toolbar>
                <Typography variant='h5' sx={{flexGrow:1}}>Weather Report</Typography>
                <form action="" onSubmit={handelSubmit}>
                    <TextField type='text' placeholder='enter country  name' onChange={handelChange}/>
                    <Button sx={{height:'3.4rem', color:'white'}} variant='contained' disabled={!input} type='submit'>Search</Button>
                </form>
            </Toolbar>
        </AppBar>
    </div>
  )
}

export default Nav