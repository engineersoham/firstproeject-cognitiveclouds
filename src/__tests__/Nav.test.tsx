import {getByRole, render, screen} from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Nav from '../component/Nav';
import React from 'react'

test('get snapshot for nav', ()=>{
    const shot = render(
        <BrowserRouter>
        <Nav/>
        </BrowserRouter>
    );
    expect(shot).toMatchSnapshot()
})

test('test for nav button', () => {
    render(
        <BrowserRouter>
        <Nav/>
        </BrowserRouter>
    );
    const btn = screen.getByRole('button');
    const inputs = screen.getByRole('textbox');

    expect(btn).toBeInTheDocument();
    expect(inputs).toBeInTheDocument()
})