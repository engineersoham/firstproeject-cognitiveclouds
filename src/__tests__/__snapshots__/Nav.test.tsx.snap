// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get snapshot for nav 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-i9gxme"
      >
        <header
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-dudo9s-MuiTypography-root"
            >
              WEATHERSTACK
            </div>
            <div
              class="css-1orzy9s"
            >
              <form
                action=""
              >
                <input
                  id="browsers"
                  name="name"
                  placeholder="enter country name"
                  style="height: 2rem; margin: 10px;"
                  type="text"
                  value=""
                />
                <button
                  style="height: 2.4rem; margin-top: 5px;"
                  type="submit"
                >
                  Search
                </button>
              </form>
            </div>
          </div>
        </header>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-i9gxme"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
        >
          <div
            class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-dudo9s-MuiTypography-root"
          >
            WEATHERSTACK
          </div>
          <div
            class="css-1orzy9s"
          >
            <form
              action=""
            >
              <input
                id="browsers"
                name="name"
                placeholder="enter country name"
                style="height: 2rem; margin: 10px;"
                type="text"
                value=""
              />
              <button
                style="height: 2.4rem; margin-top: 5px;"
                type="submit"
              >
                Search
              </button>
            </form>
          </div>
        </div>
      </header>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;